infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> Which callback are you talking about? Do you mean that we should pass RelayerPolicy to rsEnqueue and rsDequeue? I don't see any place where IO is in negative position in RelayerPolicy.

It's in Edsko's relayer, not the one used in this patch. I suppose we shouldn't be discussing this without having that program text to look at. The `RelayerPolicy` is used to create the relayer itself, which also demands a callback.